src/settings/MappingProfiles/MappingProfilesFormContainer/MappingProfilesFormContainer.js

```diff
@@ -15,6 +15,24 @@ import { MappingProfilesForm } from '../MappingProfilesForm';
 import { mappingProfileTransformations } from '../MappingProfilesTransformationsModal/TransformationsField/transformations';
 import { generateTransformationFieldsValues } from '../MappingProfilesTransformationsModal/TransformationsField';
 
+const isValidRecordTypesMatching = (selectedTransformations = [], selectedRecordTypes = []) => {
+  const intersection = {};
+
+  selectedTransformations.forEach(selectedTransformation => {
+    intersection[selectedTransformation.recordType] = selectedTransformation.recordType;
+  });
+
+  selectedRecordTypes.forEach(recordType => {
+    if (!intersection[recordType]) {
```

Comment:
I think this check is not needed. Won't the snippet below be enough to do the job?
```
if (intersection[recordType]) {
  delete intersection[recordType];
}
```